Assets/Scripts/BallHealth.cs

```diff
@@ -23,6 +24,10 @@ void Update()
     public void TakeDamage(float damage)
     {
         SetHealth(currentHealth - damage);
+        if (currentHealth <= 0)
+        {
+            SceneManager.LoadScene(SceneManager.GetActiveScene().buildIndex);
```

Comment:
- Remove SceneManagement namespace and all logic for changing Scenes.
- Create an Action that is invoked when the Ball runs out of health.
- Create a LevelManager class that listens to the ball death action and reloads the scene when the ball dies.